src/data_factory_testing_framework/functions/evaluator/expression_evaluator.py

```diff
@@ -39,7 +46,8 @@ def __init__(self) -> None:
         expression_grammar = f"""
             // TODO: add support for array index
             ?expression_start: expression_evaluation
-            expression_evaluation: expression_call [expression_object_accessor]*
+            # TODO: probably object accessor does not apply to all below in expression_evaluation
+            expression_evaluation: (expression_logical_bool | expression_branch | expression_call) ((("." EXPRESSION_PARAMETER_NAME) | EXPRESSION_ARRAY_INDEX)+)?
```

Comment:
What's the reason of merging expression_calls with expression_accessors in a single rule?

Reply:
Good remark, and I hope my memory doesn't fail me:
Now that we evaluate the three from top to bottom the `expression_object_accessor` needs to have the actual values it accesses as children. The tree composition therefore changes. There are likely multiple ways to achieve it, but I made it part of the `expression_evaluation` mostly to keep the evaluated tree as simple as possible to keep it maintainable and DRY. 

Previously, the `expression_evaluation` rule was dead simple (just a passthrough afaik) but also used to to the expression_object_accesor processing:
```python
def expression_evaluation(self, values: list[Token, str, int, float, bool, list]) -> [str, int, float, bool]:
        eval_value = values[0]

        remaining_fields = values[1:]
        return self._evaluate_expression_object_accessors(eval_value, remaining_fields)
```

Bottom line: the changes just simplify the tree, but things stay (more or less) the same.
For me unclear, but also present in the code base, is if all other rules acutally support arrays or objects, this left a TODO:
`# TODO: probably object accessor does not apply to all below in expression_evaluation`. If we, however, assume valid ADF expressions that this should not matter (for now)